backend/src/main/kotlin/blue/mild/covid/vaxx/service/IsinService.kt

```diff
@@ -104,6 +116,104 @@ class IsinService(
+            "J01"
+        else
+            patient.indication
+
+        return runCatching {
+            val isinVaccination = createVaccination(
+                IsinVaccinationCreateOrUpdateDtoIn(
+                    id = null,
+                    pacientId = patient.isinId,
+                    typOckovaniKod = "CO19",
+                    indikace = listOf(indication),
+                    indikaceJina = configuration.indikaceJina
+                )
+            )
+
+            if (isinVaccination.id == null) {
+                throw NoSuchFieldException("We expect ISIN return non null vaccination id from create vaccination api call")
+            }
+
+            logger.debug("Creating vaccination in ISIN was successful for patient with ISIN ID ${patient.isinId}.")
+            logger.debug("Data obtained from ISIN: $isinVaccination")
+
+            val dose = createVaccinationDose(
+                IsinVaccinationDoseCreateOrUpdateDtoIn(
+                    id = null,
+                    vakcinaceId = isinVaccination.id,
+                    ockovaciLatkaKod = configuration.ockovaciLatkaKod,
+                    datumVakcinace = vaccination.vaccinatedOn,
+                    typVykonuKod = null,
+                    sarze = vaccination.vaccineSerialNumber,
+                    aplikacniCestaKod = "IM",
+                    mistoAplikaceKod = if (vaccination.bodyPart == VaccinationBodyPart.DOMINANT_HAND) "DP" else "NP",
+                    expirace = vaccinationExpirationInstant,
+                    poznamka = vaccination.notes,
+                    stav = null
+                )
+            )
+
+            logger.debug("Creating vaccination dose in ISIN was successful for patient with ISIN ID ${patient.isinId}.")
+            logger.debug("Data obtained from ISIN: $dose")
+
+            logger.info("Exporting vaccination to ISIN was successful for patient with ISIN ID ${patient.isinId}.")
+            true
+        }.getOrElse {
+            logger.error(it) {
+                "Exporting vaccination to ISIN failed for patient with ISIN ID ${patient.isinId}"
+            }
+            false
+        }
+    }
+
+    private suspend fun createVaccination(vaccinationDtoIn: IsinVaccinationCreateOrUpdateDtoIn): IsinVaccinationDto {
+        val url = createIsinURL(URL_CREATE_OR_CHANGE_VACCINATION)
+        val data = if (vaccinationDtoIn.pracovnik == null)
+            vaccinationDtoIn.copy(pracovnik = configuration.pracovnik)
+        else
+            vaccinationDtoIn
+
+        logger.info { "Executing ISIN HTTP call." }
+        return isinClient.post<HttpResponse>(url) {
+            contentType(ContentType.Application.Json)
+            accept(ContentType.Application.Json)
+            body = data
+        }.receive()
+    }
+
+    private suspend fun createVaccinationDose(vaccinationDoseDtoIn: IsinVaccinationDoseCreateOrUpdateDtoIn): IsinVaccinationDoseDto {
```

Comment:
je to prakticky stejna fce jako predchozi, ale to ted neresme

Reply:
přidal jsem todo